src/utils/framePostHandler.ts

```diff
@@ -47,20 +61,76 @@ export async function framePostHandler(req: NextApiRequest) {
   const headers = new Headers();
   headers.append('Content-Type', 'text/html');
 
-  return new Response(
-    `
-      <html>
-        <meta property="fc:frame" content="vNext">
-        ${hasPrevious ? '<meta property="fc:frame:button:1" content="←">' : ''}
-        <meta property="fc:frame:button:${hasPrevious ? 2 : 1}" content="→">
-        <meta property="fc:frame:image" content="${url}">
-        <meta property="fc:frame:post_url" content="${url}">
-        <body>gm</body>
-      </html>
-    `,
-    {
-      status: 200,
-      headers,
+  const showTwoButtons = hasPrevious;
+  const frameButtons: [FrameButtonMetadata, ...FrameButtonMetadata[]] = [
+    { label: showTwoButtons ? '←' : buttonContent, action: 'post' },
+    ...(showTwoButtons ? [{ label: buttonContent, action: 'post' } as FrameButtonMetadata] : []),
+  ];
+  const image = url.toString();
+  const postUrl = url.toString();
+
+  const htmlConfig = {
+    buttons: frameButtons,
+    image: {
+      src: image,
+      aspectRatio: '1.91:1' as AllowedAspectRatio,
+    },
+    postUrl,
+  };
+
+  position = url.searchParams.get('position');
+
+  // use square aspect ratio for image if appropriate for collection token
+  // TODO(rohan): similarly support it for other types of frames
+  if (handleSquareAspectRatioType === 'CollectionFrame' && position) {
+    let squareAspectRatio = false;
+
+    const chain = url.searchParams.get('chain');
+    const contractAddress = url.searchParams.get('contractAddress');
+
+    if (!chain || typeof chain !== 'string') {
+      throw new Error('Error: chain not found');
     }
-  );
+
+    if (!contractAddress || typeof contractAddress !== 'string') {
+      throw new Error('Error: contractAddress not found');
+    }
+
+    const queryResponse = await fetchGraphql({
+      queryText: fcframeContractCommunityDimensionsOpengraphQuery,
+      variables: {
+        contractCommunityKey: {
+          contract: {
+            address: contractAddress,
+            chain,
+          },
+        },
+      },
+    });
```

Comment:
ok so this does NOT block this PR, but my concern with this is that we're effectively adding another network request to the post handler:
1. user requests initial splash screen
2. user clicks next
3. our post handler sends a request to our server to check dimensions **[NEW 🔥 ]**
4. post handler returns HTML with aspect ratio for frame to parse
5. frame requests image from another URL
6. our GET handler sends a request to our server for the image
7. image gets rendered

basically step 3 is adding additional latency / overhead. this is totally fine for now to get things working, but i wonder if there's a way to encode the "next element / prev element aspect ratios" in the original frame URL such that the client (frame viewer) knows _what aspect ratio to request_ as they click next or previous.

don't think this is trivial to do though, esp since the first splash frame that's rendered is a pure URL that comes from our main `frontend` repo that has no idea of what NFTs live inside of it